portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,70 +27,107 @@
+        // Add fun facts first to factsAndComments return variable.
+        List<String> factsAndComments = new ArrayList<>();
+        factsAndComments.addAll(funFacts);        
         
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         Query query = new Query("siteComments").addSort("timeStamp", SortDirection.DESCENDING);
         PreparedQuery results = datastore.prepare(query);
-
+ 
         List<String> pastComments = new ArrayList<>();
+        int validNumRequest = 0;
+        int numCommentsWanted = 0;
+        boolean numCommentsWantedTrigger = true;
+ 
         for (Entity entity : results.asIterable()) {
-            String comment = (String) entity.getProperty("newComment");
-            pastComments.add(comment);
+            validNumRequest = Integer.parseInt((String)(entity.getProperty("seeNumComments")));
+            if (validNumRequest >= 0){
+                String comment = (String) (entity.getProperty("newComment"));
+
+                //For testing queried result comments
+                // System.out.println("Query result comment: " + comment);
+
+                pastComments.add(comment);
+            } else {
+                System.out.println("Input of number comments wanted not greater than or equal to 0: " + validNumRequest);
+            }
+           
+            //Grab number of comments wanted to be seen by user only in the first entity
+            while (numCommentsWantedTrigger) {
+                numCommentsWanted = validNumRequest;
+                numCommentsWantedTrigger = false;
+            }
+        }
+ 
+        if (numCommentsWanted > pastComments.size()) {
+            numCommentsWanted = pastComments.size();           
         }
+ 
+        pastComments = pastComments.subList(0, numCommentsWanted);
 
-        List<String> factsAndComments =  new ArrayList<>();
-        factsAndComments.addAll(funFacts);
-        factsAndComments.addAll(pastComments);
+        // For testing proper number and right comments are in sublist
+        // for (String comment : pastComments) {
+        //     System.out.println("sublist comment: " + comment);
+        // }
 
+        factsAndComments.addAll(pastComments);
+ 
         response.setContentType("application/json;");
         Gson gson = new Gson();
         response.getWriter().println(gson.toJson(factsAndComments));
     }
-
+ 
     /** Process POST request form input, add to pastComments, and redirect to index.html */
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        // Get the input from the form.
-        String comment = getParameter(request, "text-input", "");
         long timestamp = System.currentTimeMillis();
-
+        
+        // Get the input from the form.
+        String comment = getParameter(request, "text-input", ""); 
```

Comment:
You can probably ignore this feedback for your intern project, but keep in mind it's a good idea to do clean up user text input before you store/render it.  I'm not expert in this area, but I believe as it stands this is vulnerable to an XSS attack.

Reply:
Yea that is a good point! It hasn't been mentioned in the tutorials so far for us interns.